common/components/UI/DomainsComponents/DomainModal/DomainForm/index.tsx

```diff
@@ -16,6 +18,18 @@ const DomainForm: FC<Props> = ({
   editable = true,
   setIsValueChanged,
 }) => {
+  const { data: session } = useSession()
+  const { data: user } = useGetCurrentUserQuery()
+  const [initialValue, setInitial] = useState<{ [key: string]: any }>({
+    adminEmails: [],
+  })
+  useEffect(() => {
+    if(session?.user.email && user.roles.includes("DomainAdmin")){
+      setInitial({
+        adminEmails: [session?.user.email],
+      })
+    }
+  }, [form])
```

Comment:
Can all this logic go to `EmailSelect` and run internally? To make it reusable add some prop like `{ autocomplete?: booleand = true }` than use like:

```
<EmailSelect autocomplete={user.roles.includes(Roles.DOMAIN_ADMIN)}/>
```

```
const { data: user } = useGetCurrentUserQuery()

useEffect(() => {
  if (autocomplete) {
    form.setFieldValue('adminEmails', [user?.email])
  }
}, [form, autocomplete])
```

Make sure to not broke editing with unexpected field value override. Add some mechanism to handle preview and editing form cases